File: Collectibles.Frontend/collectibles/src/types/Authentication.ts
import React from "react";

export interface AuthenticationResponse {
    accessToken: string;
    refreshToken: string;
}

interface AuthenticationRequest {
    email: string;
    password: string;
}

export interface LoginRequest extends AuthenticationRequest {
    
}

export interface SignupRequest extends AuthenticationRequest {
    username: string;
}

export interface RefreshRequest {
    refreshToken: string;
}

export type AuthState = {
    isLoading: boolean;
    isSignedIn: boolean;
    isConnectionError: boolean;
};

export type AuthAction =
    | { type: 'SIGN_OUT' }
    | { type: 'SIGN_IN' }
    | { type: 'CONNECTION_ERROR' }

export interface AuthContextType {
    state: AuthState;
    dispatch: React.Dispatch<AuthAction>;
}

type FieldState = {
    value: string;
    errors: string[];
};

export interface LoginForm {
    email: FieldState;
    password: FieldState;
    apiErrors: string[]
}

export interface SignupForm {
    email: FieldState;
    password: FieldState;
    confirmPassword: FieldState;
    username: FieldState;
    apiErrors: string[]
}

File: Collectibles.Backend/Collectibles.Backend.Api/Models/Authentication.cs
namespace Collectibles.Backend.Api.Models;

public class AuthenticationRequest
{
    public string Email { get; set; }
    public string Password { get; set; }
}
public class SignupRequest : AuthenticationRequest
{
    public string? Username { get; set; }
}

public class LoginRequest : AuthenticationRequest
{

}

public class RefreshRequest
{
    public string RefreshToken { get; set; }
}

public class AuthenticationResponse
{
    public string AccessToken { get; set; }
    public string RefreshToken { get; set; }
}

public class RefreshToken
{
    public Guid UserId { get; set; }
    public Guid TokenId { get; set; }
    public string Token { get; set; }
    public DateTime ExpiresAt { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? UserAgent { get; set; }
    public string? IpAddress { get; set; }
    public bool IsValid { get; set; }
}

public class LogoutRequest
{
    public string RefreshToken { get; set; }
}